control_server.go

```diff
@@ -210,7 +216,16 @@ func (this *ControlServer) HandleVerify(w http.ResponseWriter, r *http.Request)
 		return
 	}
 
-	http.Redirect(w, r, "/", http.StatusSeeOther)
+	returnSuccess(w, r)
+}
+
+func (this *ControlServer) HandleStatus(w http.ResponseWriter, r *http.Request) {
+	status := this.fetchStatus()
+	w.Header().Set("Content-Type", "application/json")
+	err := json.NewEncoder(w).Encode(status.Progress)
```

Comment:
We want to encode the whole `status` and not only `status.Progress` right?

Reply:
Yup, I was messing around with things and forgot to revert this change